CMakeLists.txt

```diff
@@ -27,29 +28,119 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
 set(CMAKE_POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
 
-if(FIND_PM_DEPS)
-  find_package(SwiftSystem CONFIG REQUIRED)
-  find_package(TSC CONFIG REQUIRED)
+# Toolchain Vended dependencies
+find_package(dispatch QUIET)
+find_package(Foundation QUIET)
+
+include(FetchContent)
+
+set(_SPM_SAVED_BUILD_TESTING ${BUILD_TESTING})
+set(_SPM_SAVED_BUILD_EXAMPLES ${BUILD_EXAMPLES})
+
+set(BUILD_EXAMPLES NO)
+set(BUILD_TESTING NO)
+
+find_package(ArgumentParser CONFIG)
```

Comment:
Do you need to explicitly call `find_package` first? If you don't do anything FetchContent will call `find_package` first before doing anything, and only if it's not found will it do things. To tell it to use the `CONFIG` search mode, add do
```cmake
FetchContent_Declare(ArgumentParser
  GIT_REPOSITORY https://github.com/apple/swift-argument-parser
  GIT_TAG 1.2.3
  FIND_PACKAGE_ARGS CONFIG)
```

https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_declare -- `FIND_PACKAGE_ARGS`

erm... wait, that's only available in 3.24+. That makes things more challenging.

Reply:
I think I'll leave this as is until we get a newer CMake.